include/zephyr/bluetooth/classic/a2dp.h

```diff
@@ -91,20 +533,242 @@ struct bt_a2dp;
+	 * or after it is sent on air or flushed.
+	 *
+	 * This callback is only used if the ISO data path is HCI.
+	 *
+	 * @param stream Stream object.
+	 */
+	void (*sent)(struct bt_a2dp_stream *stream);
+#endif
+};
+
+/**
+ * @brief Register Audio callbacks for a stream.
+ *
+ * Register Audio callbacks for a stream.
+ *
+ * @param stream Stream object.
+ * @param ops    Stream operations structure.
+ */
+void bt_a2dp_stream_cb_register(struct bt_a2dp_stream *stream, struct bt_a2dp_stream_ops *ops);
+
+/** @brief configure endpoint.
+ *
+ *  bt_a2dp_discover can be used to find remote's endpoints.
+ *  This function to configure the selected endpoint that is found by
+ *  bt_a2dp_discover.
+ * This function sends AVDTP_SET_CONFIGURATION.
+ *
+ *  @param a2dp The a2dp instance.
+ *  @param stream Stream object.
+ *  @param local_ep The configured endpoint that is registered.
+ *  @param remote_ep The remote endpoint.
+ *  @param config The config to configure the endpoint.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_a2dp_stream_config(struct bt_a2dp *a2dp, struct bt_a2dp_stream *stream,
+		struct bt_a2dp_ep *local_ep, struct bt_a2dp_ep *remote_ep,
+		struct bt_a2dp_codec_cfg *config);
+
+/** @brief establish a2dp streamer.
+ *
+ * This function sends the AVDTP_OPEN command and create the l2cap channel.
+ *
+ *  @param stream The stream object.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_a2dp_stream_establish(struct bt_a2dp_stream *stream);
+
+/** @brief release a2dp streamer.
+ *
+ * This function sends the AVDTP_CLOSE command and release the l2cap channel.
+ *
+ *  @param stream The stream object.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_a2dp_stream_release(struct bt_a2dp_stream *stream);
+
+/** @brief start a2dp streamer.
+ *
+ * This function sends the AVDTP_START command.
+ *
+ *  @param stream The stream object.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_a2dp_stream_start(struct bt_a2dp_stream *stream);
+
+/** @brief suspend a2dp streamer.
+ *
+ * This function sends the AVDTP_SUSPEND command.
+ *
+ *  @param stream The stream object.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_a2dp_stream_suspend(struct bt_a2dp_stream *stream);
+
+/** @brief re-configure a2dp streamer
+ *
+ * This function sends the AVDTP_RECONFIGURE command.
+ *
+ *  @param stream The stream object.
+ *  @param config The config to configure the stream.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_a2dp_stream_reconfig(struct bt_a2dp_stream *stream, struct bt_a2dp_codec_cfg *config);
+
+/** @brief get the stream l2cap mtu
+ *
+ *  @param stream The stream object.
+ *
+ *  @return mtu value
+ */
+uint32_t bt_a2dp_get_mtu(struct bt_a2dp_stream *stream);
+
+#if ((defined(CONFIG_BT_A2DP_SOURCE)) && (CONFIG_BT_A2DP_SOURCE > 0U))
```

Comment:
> Suggest change to `#if CONFIG_BT_A2DP_SOURCE`

Will that compile if `CONFIG_BT_A@DP_SOURCE` is undefined (which it will be when you set the Kconfig option to `n`)

Reply:
Yes. It is a header file. 
Maybe it could be `#if defined(CONFIG_BT_A2DP_SOURCE)`.

Reply:
Updated